oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -225,7 +226,12 @@ public Query baseQuery() {
                 hasTieBreaker = true;
             } else if (JCR_SCORE.equals(sortPropertyName)) {
                 fieldName = "_score";
-            } else if (indexProperties.containsKey(sortPropertyName)) {
+            } else if (indexProperties.containsKey(sortPropertyName) || elasticIndexDefinition.getDefinedRules()
+                    .stream().map(rule -> rule.getConfig(sortPropertyName))
+                    .collect(Collectors.toList()).size() > 0) {
```

Comment:
this should produce the same result and should be more robust (better handling of nulls). I have not tested it
```suggestion
                    .stream().map(rule -> Optional.ofNullable(rule.getConfig(sortPropertyName))).findAny().isPresent()) {
```

Reply:
`.stream().map(rule -> Optional.ofNullable(rule.getConfig(sortPropertyName))).findAny().isPresent()`
This seems to be always returning true. Looks like `map(rule -> Optional.ofNullable(rule.getConfig(sortPropertyName)))` returns an Optional[Optional.Empty] so the element as result of findAny() is actually an Optional of another Optional  (Optional.Empty) and isPresent returns true in this case.

Looking at this again - 
`.stream().map(rule -> rule.getConfig(sortPropertyName))
                    .collect(Collectors.toList()).size() > 0)` this too would always return true since the size of this list would always be greater than 0 even if all the elements are null.